src/auto/AutoModule.lua

```diff
@@ -55,11 +55,13 @@ return function(moduleName)
       end
 
       function module:accGradParameters(input, gradOutput, scale)
-         if not self.grads then
-            self.grads = self.b({input=input, weight=self.weight, bias=self.bias}, gradOutput)
+         if self.weight then
```

Comment:
For the PR, a test that tests the full set of combos ((weight, bias),
(weight), (bias), ()) would be best.

On Thu, Jan 21, 2016 at 6:17 PM, Francisco Massa <notifications@github.com>
wrote:

> In src/auto/AutoModule.lua
> <https://github.com/twitter/torch-autograd/pull/86#discussion_r50480627>:
>
> > @@ -55,11 +55,13 @@ return function(moduleName)
> >        end
> >
> >        function module:accGradParameters(input, gradOutput, scale)
> > -         if not self.grads then
> > -            self.grads = self.b({input=input, weight=self.weight, bias=self.bias}, gradOutput)
> > +         if self.weight then
>
> I think you need to be more careful with that, there is at least one
> module in nn which has bias but no weight, see
> https://github.com/torch/nn/blob/master/Add.lua#L9
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/twitter/torch-autograd/pull/86/files#r50480627>.
>